projects/graph/social/graph.py

```diff
@@ -0,0 +1,247 @@
+"""
+Simple graph implementation
+"""
+"""
+{
+'0': {'1', '3'},
+'1': {'0'},
+'2': {set()},
+'3': {'0'}
+}
+"""
+import pprint
+from queue import *
+
+
+class Vertex:
+    def __init__(self, name, neighbors=list(), distance=9999, color='black'):
+        self.name = name #label
+        self.neighbor = list() #same as edges?
+        self.distance = 9999 #destination
+        self.color = 'black' #black for visited, red for not visited
+
+
+
+class Graph:
+    """Represent a graph as a dictionary of vertices mapping labels to edges."""
+    def __init__(self):
+        self.vertices = {}
+        self.edges = set()
+        # self.destination = destination
+        # self.label = label
+
+
+
+    def add_vertex(self, vertex):
+        self.vertices[vertex] = set()
+        print(self.vertices)
+
+    def add_edge_bi(self, vertex_start, vertex_end):
+        if vertex_start not in self.vertices:
+            print(f'You chose: {vertex_start}. That Beginning Vertex does not exist...')
+            return
+        if vertex_end not in self.vertices:
+            print(f"You chose: {vertex_end} That End Vertex does not exist...")
+            return
+        for i in self.vertices:
+            # temp = set()
+            # temp.add(i)
+            if i == vertex_start:
+                self.vertices[vertex_start].add(vertex_end)
+                print(self.vertices)
+            if vertex_end == i:
+                self.vertices[vertex_end].add(vertex_start)
+                print(self.vertices)
+
+    def add_edge_mono(self, vertex_start, vertex_end):
+        if vertex_start not in self.vertices:
+            print(f'You chose: {vertex_start}. That Beginning Vertex does not exist...')
+            return
+        if vertex_end not in self.vertices:
+            print(f"You chose: {vertex_end} That End Vertex does not exist...")
+            return
+        for i in self.vertices:
+            # temp = set()
+            # temp.add(i)
+            if i == vertex_start:
+                self.vertices[vertex_start].add(vertex_end)
+                print(self.vertices)
```

Comment:
I like that you mix unidirectional and bidirectional edges in the same graph. Very cool.